File: cpp/136_Single_Number.cpp
// 136. Single Number
// https://leetcode.com/problems/single-number/
// Difficulty: Easy | Language: cpp | Runtime: 0 ms | Memory: 20.7 MB

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int result = 0; 
        for (int i = 0; i < nums.size(); i++){
            result ^= nums[i];
        }
        return result;
    }
};

File: java/136_Single_Number.java
// 136. Single Number
// https://leetcode.com/problems/single-number/
// Difficulty: Easy | Language: java | Runtime: 248 ms | Memory: 40.2 MB

class Solution {
    public int singleNumber(int[] nums) {
        int n = nums.length;
        for (int i = 0; i < n; i++) { 
            int j; 
            for (j = 0; j < n; j++) 
                if (i != j && nums[i] == nums[j]) 
                    break; 
            if (j == n) 
                return nums[i]; 
        }
        return -1;
    }
}
